server/task/task-distributed/src/main/scala/org/apache/james/task/eventsourcing/cassandra/CassandraTaskExecutionDetailsProjectionDAO.scala

```diff
@@ -62,23 +60,34 @@ class CassandraTaskExecutionDetailsProjectionDAO @Inject()(session: CqlSession,
 
   private val listStatement = session.prepare(selectFrom(TABLE_NAME).all().build())
 
+  private val listBeforeDateStatement : PreparedStatement = session.prepare(selectFrom(TABLE_NAME)
+    .all()
+    .allowFiltering()
+    .whereColumn(SUBMITTED_DATE).isLessThanOrEqualTo(bindMarker(SUBMITTED_DATE))
+    .build())
+
+  private val removeStatement : PreparedStatement = session.prepare(deleteFrom(TABLE_NAME)
+    .whereColumn(TASK_ID).isEqualTo(bindMarker(TASK_ID))
+    .whereColumn(SUBMITTED_DATE).isEqualTo(bindMarker(SUBMITTED_DATE))
+    .build())
+
```

Comment:
IMO Using listStatement and doing application side filtering is good enough for now.